indico/modules/events/editing/controllers/backend/timeline.py

```diff
@@ -50,17 +50,20 @@ def get_file_context(self):
         return 'event', self.event.id, 'editing', self.contrib.id, self.editable_type.name
 
 
-class RHEditingUploadPaperLastRevision(RHEditingUploadFile):
+class RHEditingUploadContributionFile(RHEditingUploadFile):
     @use_kwargs({
         'id': fields.Int()
     })
     def _process(self, id):
-        last_rev = self.contrib.paper.get_last_revision()
-        if last_rev:
-            found = next((f for f in last_rev.files if f.id == id), None)
-            if found:
-                with found.open() as stream:
-                    return self._save_file(found, stream)
+        files = []
+        if self.contrib.editables:
+            files = [file.file for e in self.contrib.editables
+                     if e.type == self.editable.type for file in e.revisions[-1].files]
+        if self.contrib.paper and (last_rev := self.contrib.paper.get_last_revision()):
+            files.extend(last_rev.files)
+        if found := next((f for f in files if f.id == id), None):
```

Comment:
`EditingRevisionFile` and `PaperFile` are in different tables, so it's possible for an ID collision between them -> as discussed on Mattermost we need e.g. a query param indicating the source of the file

Reply:
It was a bit more complicated than foreseen, but we're past it.